src/components/List.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+    <div>
+        <div v-if="isEditMode" class="addArea">
+            <p>Edit mode is activated!</p>
+            <input text="text" v-model="updatedCity.name" />
+            <button @click="changeMode(),updateCity(updatedCity)">Save</button>
```

Comment:
Whenever you have a complex logic, it is recommended to use a method instead, and do all the logic there :)